protocol/client/contact.go

```diff
@@ -56,6 +58,32 @@ type Contact struct {
 	PublicKey *ecdsa.PublicKey `json:"-"`
 }
 
+// CreateContactPrivate creates a new private contact.
+func CreateContactPrivate(name, pubKeyHex string) (c Contact, err error) {
+	pubKeyBytes, err := hexutil.Decode(pubKeyHex)
+	if err != nil {
+		return
+	}
+
+	c.Name = name
+	c.Type = ContactPrivate
+	c.State = ContactAdded
```

Comment:
i think it should be ContactNew by default. unless contact is added by user

Reply:
Good point, I will add an argument. Currently it's only used when adding a contact by the user, though.